proposals/nnnn-sdl_on_qnx700_agl.md

```diff
@@ -0,0 +1,254 @@
+
+#### Modification in test scripts
+
+Some scripts should be modified to use SDL on remote workstation.  
+All operations with SDL files ( hmi_capabilities, preloaded_pt, etc ...) should be covered with wrappers that support either local or remote execution. 
+
+
+## Testing Approach
+
+The following items should be checked for all these platforms :
+ 
+ - Compilation
+ - Unit tests
+ - Automated smoke
+ - Existing features automated test cases (in case if feature is applicable on the virtual workstation)  
+
+SDL will be tested on virtual workstation for each supported platform. 
+For communication with mobile will be used **only** TCP transport.
+For communication with HMI will be used TCP/WebSockets transport.
+
+All supported platforms (Ubuntu, AGL, QNX) will share codebase for communication with mobile and HMI.  
+
+#### Manual testing
+
+Manual testing will be performed using WebHMI https://github.com/smartdevicelink/sdl_hmi
+No changes in WebHMI expected, WebHMI will be executed on real developer workstation. 
+
+Mobile application may be connected to SDL via TCP.
+
+### Automated testing.
+
+With some changes in ATF test framework and existing scripts SDL will be tested automatically on all 3 platforms. 
+
+The same test scripts will be executed for : 
+ - Ubuntu 
+ - AGL
+ - QNX
+ 
+Test coverage will include:
+  - smoke testing of SDL (basic SDL scenarios and 3 policy flows) 
+  - all delivered features that applicable for cross platform testing.
+  
+Each further delivered feature should be tested on all supported platforms.
+
+
+## Potential downsides
+
+N/A
+
+## Impact on existing code
+
+SDL core will be modified in platform specific places.
+All components that uses **utils** will be affected and need to be retested on target platform. 
+ 
+## Alternatives considered
+
+
+#### Additional transport support on AGL, QNX virtual workstation
+
+Add USB and Bluetooth transport support for virtualized QNX and AGL platforms.
+These transport adapters will not be used for real OEMs, the only place to use them is virtual workstations.
+Any way this transport communication won't be real, because SDL run on virtual workstation. 
+Supporting of additional transports on virtual workstation is rather exhausted. There is no valuable reason to keep supporting many transports on multiple platforms except as best practice example for OEM suppliers. 
+
+Pros : 
+ - Testing may be done via additional transports
+
+Cons:
+- Big efforts for automation of transport testing 
+- Big efforts for supporting many transports on multiple platform
+- Transport specific code will not be actual for OEM suppliers
+- Transport testing will be executed through virtualisation layer.
+
+#### Using real hardware with QNX and AGL
+
+SDL may be tested not on virtual workstation, but on real hardware with QNX, Ubuntu and AGL.
```

Comment:
Change to:
SDL may not be tested on virtual workstation, but on real hardware with QNX, Ubuntu and AGL.